src/Microsoft.ML.LightGBM/LightGbmTrainerBase.cs

```diff
@@ -3,14 +3,12 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-using System.Linq;
-using System.Threading;
-using System.Threading.Tasks;
 using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
```

Comment:
do you really need linq?

Reply:
yeah there are references to it in this file. Last()

---
In reply to: [199288120](https://github.com/dotnet/machinelearning/pull/454#discussion_r199288120) [](ancestors = 199288120)

Reply:
Reactivating. This is actually pretty bad. Change that code to get the second to last index. I know it's not your fault, but, let's not let bad code stand, if we know where it is.

---
In reply to: [199289028](https://github.com/dotnet/machinelearning/pull/454#discussion_r199289028) [](ancestors = 199289028,199288120)